server/cluster/cluster_worker.go

```diff
@@ -45,6 +45,9 @@ func (c *RaftCluster) HandleRegionHeartbeat(region *core.RegionInfo) error {
 
 // HandleAskSplit handles the split request.
 func (c *RaftCluster) HandleAskSplit(request *pdpb.AskSplitRequest) (*pdpb.AskSplitResponse, error) {
+	if c.GetUnsafeRecoveryController() != nil && c.GetUnsafeRecoveryController().IsRunning() {
+		return nil, errors.New("Unsafe recovery is running, split is paused")
```

Comment:
return error with unknown type in the response header instead of grpc error

Reply:
What's the consideration behind this? I did a little bit code search, seems that header errors are not widely using?

Reply:
TiKV may panic on the grpc error, but will pass through the error in the header. https://github.com/pingcap/tikv/blob/5c48a7a7a540d45fe301da193ce06aeb2cee00b9/components/pd_client/src/client.rs#L701

Reply:
@Connor1996 @rleungx Please take a look at the proposed code changes in `server/grpc_service.go`.